src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/CVImageCriteriaEvaluator.cs

```diff
@@ -0,0 +1,342 @@
+            var resultList = new List<string>();
+
+            ConcurrentDictionary<int,List<CVImageResult>> cvImageResults = null;
+            List<string> priorResults = null;
+            lock (_requestTracker)
+            {
+                _resultTracker.TryGetValue(segmentNumber, out cvImageResults);
+                _priorResultsTracker.TryGetValue(segmentNumber, out priorResults);
+                var resultsString = cvImageResults == null ? "null":$"\n[{string.Join(",\n", cvImageResults.Select(pair => $"{pair.Key}:[{(pair.Value==null?"null":string.Join(",\n", pair.Value))}]"))}]";
+                if (cvImageResults != null)
+                {
+                    RGDebug.LogDebug($"CVImageCriteriaEvaluator - Matched - botSegment: {segmentNumber} - cvImageResults: {resultsString}");
+                }
+            }
+
+            var criteriaListCount = criteriaList.Count;
+
+            if (priorResults == null && cvImageResults != null && cvImageResults.Count == criteriaListCount && cvImageResults.All(cv => cv.Value != null))
+            {
+                // Evaluate the results if we have them all for each criteria
+                for (var i = criteriaListCount - 1; i >= 0; i--)
+                {
+                    var criteria = criteriaList[i];
+                    var found = false;
+                    if (!criteria.transient || !criteria.Replay_TransientMatched)
+                    {
+                        var criteriaData = criteria.data as CVImageKeyFrameCriteriaData;
+
+                        if (cvImageResults.TryGetValue(i, out var cvImageResultList) && cvImageResultList is { Count: > 0 })
+                        {
+                            var withinRect = criteriaData.withinRect;
+                            // we had the result for this criteria
+                            if (withinRect != null)
+                            {
+                                foreach (var cvImageResult in cvImageResultList)
+                                {
+                                    // ensure result rect is inside
+                                    var relativeScaling = new Vector2(withinRect.screenSize.x / (float)cvImageResult.resolution.x, withinRect.screenSize.y / (float)cvImageResult.resolution.y);
+
+                                    // check the bottom left and top right to see if it intersects our rect
+                                    var bottomLeft = new Vector2Int(Mathf.CeilToInt(cvImageResult.rect.x * relativeScaling.x), Mathf.CeilToInt(cvImageResult.rect.y * relativeScaling.y));
+                                    var topRight = new Vector2Int(bottomLeft.x + Mathf.FloorToInt(cvImageResult.rect.width * relativeScaling.x), bottomLeft.y + Mathf.FloorToInt(cvImageResult.rect.height * relativeScaling.y));
+
+                                    // we currently test overlap, should we test fully inside instead ??
```

Comment:
This is probably fine for now, we can always add multiple modes if someone requests it